t/90http3-graceful-shutdown.t

```diff
@@ -0,0 +1,131 @@
+use strict;
+use warnings;
+use Digest::MD5 qw(md5_hex);
+use File::Temp qw(tempdir);
+use Net::EmptyPort qw(empty_port wait_port);
+use POSIX ":sys_wait_h";
+use Test::More;
+use t::Util;
+
+# test scenario:
+# 0. set up a probe for client to emit message on GOAWAY reception
+# 1. send SIGTERM to server to generate GOAWAY
+# 2. upon receiving GOAWAY, tracer will emit message
+# 3. make sure an expected string (from the GOAWAY probe) appears in the tracer log
+
+check_dtrace_availability();
+
+plan skip_all => 'curl not found'
+    unless prog_exists('curl');
+
+plan skip_all => 'mruby support is off'
+	unless server_features()->{mruby};
+
+my $tempdir = tempdir(CLEANUP => 1);
+
+my $client_prog = bindir() . "/h2o-httpclient";
+plan skip_all => "$client_prog not found"
+    unless -e $client_prog;
+
+my $quic_port = empty_port({
+    host  => "127.0.0.1",
+    proto => "udp",
+});
+
+# spawn a simple HTTP/3 echo server, excerpted from 40http3.t
+my $server = spawn_h2o(<< "EOT");
+listen:
+  type: quic
+  port: $quic_port
+  ssl:
+    key-file: examples/h2o/server.key
+    certificate-file: examples/h2o/server.crt
+http3-graceful-shutdown-timeout: 1
+hosts:
+  default:
+    paths:
+      /echo:
+        mruby.handler: |
+          Proc.new do |env|
+            [200, {}, env["rack.input"]]
+          end
+EOT
+
+wait_port({port => $quic_port, proto => 'udp'});
+
+# launch httpclient
+my $client_pid = fork;
+die "fork failed:$!"
+	unless defined $client_pid;
+
+if ($client_pid == 0) {
+	close STDERR;
+	open STDERR, ">", "/dev/null";
+	my @args = ("$client_prog", qw(-3 -t 5 -d 1000 -b 10 -c 2 -i 1000), "https://127.0.0.1:$quic_port/echo");
+	exec @args;
+	die "should not reach here!";
+}
+
+# spawn bpftrace/dtrace to probe GOAWAY frame reception
+my $tracer_pid = fork;
+die "fork(2) failed:$!"
+	unless defined $tracer_pid;
+if ($tracer_pid == 0) {
+	# child process, spawn bpftrace
+	close STDOUT;
+	open STDOUT, ">", "$tempdir/trace.out"
+		or die "failed to create temporary file:$tempdir/trace.out:$!";
+	if ($^O eq 'linux') {
+		# because there is no easy way to inspect the payload, we inspect the length of the payload instead
+		# as a minimal validation
+		exec qw(bpftrace -v -B none -p), $client_pid, "-e", <<'EOT';
+usdt::h2o:h3_frame_receive { if (arg0 == 7) { printf("H3 GOAWAY frame received: len=%d\n", arg2); } }
+EOT
+		die "failed to spawn bpftrace:$!";
+	} else {
+		exec(
+			qw(unbuffer dtrace -p), $client_pid, "-n", <<'EOT',
+:h2o::h3_frame_receive {
```

Comment:
The 2nd element denotes the process name. In this case, it should be `h2o-httpclient`.

Reply:
💯 Finally it all made sense! Thanks!